### ISSUE TITLE
URI malformed error during base64 encoding

### ISSUE THREAD

**Issue by sanex3339** [javascript-obfuscator/javascript-obfuscator#530]
```
var foo = '\ud801\ud804-';
```
when encodeUriComponent

**Comment by sanex3339** [comment]
Anyone have a idea how to fix this when this string is passed to `encodeUriComponent'?

**Comment by zamotkin** [comment]
I can review today.

**Comment by zamotkin** [comment]
@sanex3339 I can't understand how did you receive such a string, but you can try to strip invalid surrogate pairs by something like this:

```
function stripUnmatchedSurrogates (str) {
    return str.replace(/[\uD800-\uDBFF](?![\uDC00-\uDFFF])/g, '').split('').reverse().join('').replace(/[\uDC00-\uDFFF](?![\uD800-\uDBFF])/g, '').split('').reverse().join('');
}
```

**Comment by cliffhall** [comment]
I ran into a similar problem, and was getting `URI Malformed error` when doing a production build of my react app using webpack-obfusicator. I have two apps actually and one did not have the problem but the other did. In the app that did, I was basically redoing the configuration that I had on the one that worked. 

When I hit the error, I just started commenting out likely suspects in the JavascriptObfusicator configuration. I found quickly that it was `stringArrayEncoding: "base64"` causing the issue. The 'fix' turned out to be setting ` splitStrings: false`.

Cheers, and thanks for such an awesome tool!

**Comment by sanex3339** [comment]
Hi. @cliffhall can you show a small example here https://obfuscator.io/ with enabled `stringArrayEncoding` and `splitStrings`?

**Comment by cliffhall** [comment]
Unfortunately it’s a large app and I have no idea what specifically broke and caused the error.

**Comment by msheakoski** [comment]
Here is how to reproduce:

1. Visit https://obfuscator.io
2. Use the following code for the input: `"12😃45"`
3. Change String Array Threshold to: 1
4. Change String Array Encoding to: Base64
5. Enable: Split Strings
6. Change Split Strings Chunk Length to: 3
7. Click the Obfuscate button

It looks like the combination of "Split Strings Chunk Length" and "Base64" is not correctly handling multibyte characters if they occur on a chunk boundary. `😃` is composed of 2 unicode sequences, `\uD83D` and `\uDE03`. With a chunk length of 3, this particular string ends up being split into `12\uD83D` and `\uDE0345`

**Comment by sanex3339** [comment]
Ok. Will fix it using https://github.com/sallar/stringz

**Comment by sanex3339** [comment]
Fixed in `2.4.2`
